src/mame/sinclair/chloe.cpp

```diff
@@ -118,33 +119,54 @@ void chloe_state::update_memory()
 	const bool ext = BIT(m_port_ff_data, 7); // 0 - DOC 7xxxx=28+; 1 - EXT 6xxxx=24+
 	m_bank0_view.disable();
 	m_bank1_view.disable();
+	const u8 divmmc_sram_page = BIT(m_divmmc_ctrl, 0, 6);
+	const bool divmmc_sram_page_is_valid = BIT(divmmc_sram_page, 4, 2) == 0;
+	const bool mapram_mode = BIT(m_divmmc_ctrl, 6);
+	const bool conmem = BIT(m_divmmc_ctrl, 7);
+	const bool divmmc_rom_active = m_divmmc_paged || conmem;
 	for (auto i = 0; i < 8; ++i)
 	{
-		const bool paged = BIT(m_port_f4_data, i);
+		const bool paged = BIT(m_timex_mmu, i);
 
 		u8 pg;
-		if (i == 0 && (m_divmmc_paged || !paged))
+		if (i == 0 && (divmmc_rom_active || !paged))
 		{
-			m_bank0_view.select(0);
-			if (m_divmmc_paged)
-				pg = 12;
+			if (divmmc_rom_active)
+			{
+				if (!mapram_mode || conmem)
+					pg = 24;
```

Comment:
The rest of this file looks to be in Allman brace style, so this if should also have braces.